adal/tests/Test.ADAL.NET.UIAutomation/CoreMobileADALTests.cs

```diff
@@ -37,37 +40,131 @@ namespace Test.ADAL.NET.UIAutomation
+            SetInputData(controller, UiTestConstants.UiAutomationTestClientId, UiTestConstants.MSGraph);
+            PerformSignInFlow(controller, user);
+        }
+
+        private static IUser prepareForAuthentication(ITestController controller, UserQueryParameters userParams)
+        {
+            //Navigate to second page
+            controller.Tap(UiTestConstants.secondPageID);
 
             //Clear Cache
-            controller.Tap("clearCache");
+            controller.Tap(UiTestConstants.ClearCacheID);
+
+            //Get User from Lab
+            return controller.GetUser(userParams);
+        }
+
+        private static void SetInputData(ITestController controller, string ClientID, string Resource)
+        {
+            //Enter ClientID
+            controller.EnterText(UiTestConstants.clientIdEntryID, ClientID, false);
+            controller.DismissKeyboard();
+
+            //Enter Resource
+            controller.EnterText(UiTestConstants.resourceEntryID, Resource, false);
+            controller.DismissKeyboard();
+        }
+
+        private static void PerformSignInFlow(ITestController controller, IUser user)
+        {
+            string passwordInputID = string.Empty;
+            string signInButtonID = string.Empty;
+
+            switch (user.FederationProvider)
+            {
+                case FederationProvider.AdfsV4:
+                    passwordInputID = UiTestConstants.AdfsV4WebPasswordID;
+                    signInButtonID = UiTestConstants.AdfsV4WebSubmitID;
+                    break;
+                default:
+                    passwordInputID = UiTestConstants.WebPasswordID;
+                    signInButtonID = UiTestConstants.WebSubmitID;
+                    break;
+            }
 
             //Acquire token flow
-            controller.Tap("acquireToken");
+            controller.Tap(UiTestConstants.AcquireTokenID);
             //i0116 = UPN text field on AAD sign in endpoint
-            controller.EnterText("i0116", user.Upn, true);
+            controller.EnterText(UiTestConstants.WebUPNInputID, 20, user.Upn, true);
             //idSIButton9 = Sign in button
-            controller.Tap("idSIButton9", true);
+            controller.Tap(UiTestConstants.WebSubmitID, true);
             //i0118 = password text field
-            controller.EnterText("i0118", ((LabUser)user).GetPassword(), true);
-            controller.Tap("idSIButton9", true);
+            controller.EnterText(passwordInputID, ((LabUser)user).GetPassword(), true);
+            controller.Tap(signInButtonID, true);
+        }
+
+        private static void VerifyResult(ITestController controller)
+        {
+            //There may be a delay in the amount of time it takes for an authentication request to complete.
+            //Thus this method will check the result once a second for 10 seconds.
+            bool done = false;
+            int attempts = 0;
+            int maximumAttempts = 20;
```

Comment:
See if you can encapsulate a retry strategy into some retry helper and pull those constants somewhere. For inspiration https://alastaircrabtree.com/implementing-a-simple-retry-pattern-in-c/